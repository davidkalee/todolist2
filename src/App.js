import React, { Component } from 'react';
import logo from './logo.svg';
import Entry from './Entry';
import './App.css';

class App extends Component {
  constructor() {
    super();

    this.state = {
      input: '',
      storage: []
    }
  }

  updateStorage = () => {
    let { input, storage } = this.state;
    storage.push(input);
    

    this.setState({ input: '', storage }, () => console.log(this.state.input));
  }

  startTimer = () => {
    setTimeout(this.endTimer, 5000)
  }

  endTimer = () => {
    alert('TIME IS UP');
  }

  render() {
    return (
      <div className="App">
        <h2>TO DO LIST</h2>
        <hr/>
        <button id="timer" onClick={this.startTimer}>Timer</button>
        <input className="input-entry" onChange={e => this.setState({ input: e.target.value })}
        value={this.state.input}
        />
        <button
          className="submit-entry"
          onClick={this.updateStorage}
        >Submit</button>


        {
          this.state.storage.map((item, index) => {
            return <Entry key={index} entry={item} />
          })
        }

      </div>
    );
  }
}

export default App;
