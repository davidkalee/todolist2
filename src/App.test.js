import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  const app = shallow(<App />);

  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  })

  it('initializes input field of state with empty string', () => {
    expect(app.state().input).toEqual('');
  })

  it('initializes storage array in state with empty array', () => {
    expect(app.state().storage).toEqual([]);
  })

  describe('adding an entry', () => {
    const entry1 = 'buy groceries';
    it('updates input in state to match field', () => {
      app.find('.input-entry').simulate('change', { target: { value: entry1 }});
      expect(app.state().input).toEqual(entry1);
    })

    it('updates storage array in state when submitted', () => {
      app.find('.submit-entry').simulate('click');
      expect(app.state().input).toEqual('');

      expect(app.state().storage).toHaveLength(1);
      expect(app.state().storage).toEqual([entry1])
    })

    it('renders new child component', () => {
      const entry = app.find('Entry');
      expect(entry.exists()).toBe(true);
    })
  })

  describe('timer', () => {
    jest.useFakeTimers();
    
    it('calls a function after timer finishes', () => {
      const mockCallback = app.instance().endTimer = jest.fn();
      
      app.find('#timer').simulate('click');
      expect(mockCallback).not.toHaveBeenCalled();
      jest.runAllTimers();
      expect(mockCallback).toHaveBeenCalled();

    })
  })
})