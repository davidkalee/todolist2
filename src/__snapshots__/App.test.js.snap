// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          TO DO LIST
</h2>,
        <hr />,
        <button
          id="timer"
          onClick={[Function]}
>
          Timer
</button>,
        <input
          className="input-entry"
          onChange={[Function]}
          value=""
/>,
        <button
          className="submit-entry"
          onClick={[Function]}
>
          Submit
</button>,
        Array [],
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TO DO LIST",
        },
        "ref": null,
        "rendered": "TO DO LIST",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Timer",
          "id": "timer",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Timer",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-entry",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "submit-entry",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            TO DO LIST
</h2>,
          <hr />,
          <button
            id="timer"
            onClick={[Function]}
>
            Timer
</button>,
          <input
            className="input-entry"
            onChange={[Function]}
            value=""
/>,
          <button
            className="submit-entry"
            onClick={[Function]}
>
            Submit
</button>,
          Array [],
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TO DO LIST",
          },
          "ref": null,
          "rendered": "TO DO LIST",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Timer",
            "id": "timer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Timer",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-entry",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "submit-entry",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
